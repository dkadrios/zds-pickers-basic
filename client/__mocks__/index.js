export inputControls from './inputControls'
export mappings from './mappings'
export midi from './midi'
export shiftGroups from './shiftGroups'
export shifter from './shifter'
export store from './store'
export user from './user'
export version from './version'
