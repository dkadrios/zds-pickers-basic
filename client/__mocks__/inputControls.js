export default [
  {
    controlId: 0,
    channel: 10,
    calibrationLow: 0,
    calibrationHigh: 127,
    curve: 0,
    polarity: 0,
    latching: 0,
    status: 11,
    value: 110,
    active: true,
    lit: true,
  },
  {
    controlId: 1,
    channel: 10,
    calibrationLow: 0,
    calibrationHigh: 127,
    curve: 0,
    polarity: 0,
    latching: 1,
    status: 11,
    value: 111,
    active: false,
    lit: false,
  },
  {
    controlId: 2,
    channel: 10,
    calibrationLow: 0,
    calibrationHigh: 127,
    curve: 0,
    polarity: 0,
    latching: 0,
    status: 9,
    value: 110,
    active: true,
    lit: false,
  },
  {
    controlId: 3,
    channel: 10,
    calibrationLow: 0,
    calibrationHigh: 127,
    curve: 0,
    polarity: 0,
    latching: 0,
    status: 9,
    value: 110,
    active: true,
    lit: true,
  },
  {
    controlId: 4,
    channel: 10,
    calibrationLow: 0,
    calibrationHigh: 127,
    curve: 0,
    polarity: 0,
    latching: 0,
    status: 9,
    value: 110,
    active: true,
    lit: true,
  },
  {
    controlId: 5,
    channel: 10,
    calibrationLow: 0,
    calibrationHigh: 127,
    curve: 0,
    polarity: 0,
    latching: 0,
    status: 9,
    value: 110,
    active: true,
    lit: false,
  },
]
