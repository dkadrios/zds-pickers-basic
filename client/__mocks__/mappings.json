{
  "dialogVisible": false,
  "userDialogVisible": false,
  "userImportDialogVisible": false,
  "channels": [
    null,
    null,
    null,
    null,
    null,
    null,
    null,
    null,
    null,
    "BFD3",
    null,
    null,
    null,
    null,
    null,
    null
  ],
  "stockMappings": [
    "2box",
    "Addictive Drums",
    "Alesis DM Pro",
    "Alesis DM10",
    "Alesis DM5",
    "BFD2",
    "BFD3",
    "Battery 3",
    "E-MU Protean Acoustic",
    "E-MU Protean Bread n Butter",
    "E-MU Protean Studio",
    "E-MU Protean Synth",
    "EZ Drummer The Classic EZX",
    "EZDrummer",
    "EZDrummer Jazz Kit",
    "EZDrummer Latin Percussion",
    "General MIDI",
    "Infinite Instruments",
    "John Emrich - Articulations 1",
    "Kontakt 2",
    "Ocean Way Drums",
    "Pearl Redbox",
    "Roland HandSonic 15",
    "Roland TD-10",
    "Roland TD-10 Perc 1",
    "Roland TD-10 Perc 2",
    "Roland TD-10 Perc 3",
    "Roland TD-10 Perc 4",
    "Roland TD-12",
    "Roland TD-12 Perc 1 (Latin)",
    "Roland TD-12 Perc 2 (Indian)",
    "Roland TD-12 Perc 3 (African)",
    "Roland TD-12 Perc 4 (Salsa)",
    "Roland TD-12 Perc 5 (SFX)",
    "Roland TD-12 Perc 6-8 (User)",
    "Roland TD-20",
    "Roland TD-20 Perc",
    "Roland TD-3",
    "Roland TD-30",
    "Roland TD-30 Perc 1 (Latin)",
    "Roland TD-30 Perc 2 (Indian)",
    "Roland TD-30 Perc 3 (African)",
    "Roland TD-30 Perc 4 (Salsa)",
    "Roland TD-30 Perc 5 (SFX)",
    "Roland TD-30 Perc 6-8 (User)",
    "Roland TD-5",
    "Roland TD-6",
    "Roland TD-6 Perc 1",
    "Roland TD-6 Perc 10 (Special)",
    "Roland TD-6 Perc 2",
    "Roland TD-6 Perc 3 (Room)",
    "Roland TD-6 Perc 4 (Power)",
    "Roland TD-6 Perc 5 (Elec)",
    "Roland TD-6 Perc 6 (808)",
    "Roland TD-6 Perc 7 (Jazz)",
    "Roland TD-6 Perc 8 (Brush)",
    "Roland TD-6 Perc 9 (Perc)",
    "Roland TD-7",
    "Roland TD-8",
    "Roland TD-8 Perc 1",
    "Roland TD-8 Perc 10 (Special)",
    "Roland TD-8 Perc 2",
    "Roland TD-8 Perc 3 (Room)",
    "Roland TD-8 Perc 4 (Power)",
    "Roland TD-8 Perc 5 (Elec)",
    "Roland TD-8 Perc 6 (808)",
    "Roland TD-8 Perc 7 (Jazz)",
    "Roland TD-8 Perc 8 (Brush)",
    "Roland TD-8 Perc 9 (Perc)",
    "Roland TD-9",
    "Roland TD-9 Perc",
    "Simmons SD1000",
    "Simmons SD5X",
    "Simmons SD7",
    "Simmons SD7PK",
    "Simmons SD9K",
    "Sonic Reality iMap Extended",
    "Sonic Reality iMap Standard",
    "Studio ProFiles",
    "Superior Drummer",
    "Yamaha DTX 700",
    "Yamaha DTXplorer",
    "Zendrum Stompblock",
    "Zendrum TRS",
    "Zendrum TRS 2.0"
  ],
  "userMappings": ["BFD3 Custom ZenEdit", "reaper"]
}
