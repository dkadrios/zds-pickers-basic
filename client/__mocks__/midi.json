{
  "devices": [
    {
      "id": "-1325431060",
      "manufacturer": "Zendrum Studio",
      "name": "ZDS Shifter",
      "type": "output",
      "version": "",
      "state": "connected",
      "connection": "open"
    },
    {
      "id": "-312145282",
      "manufacturer": "Zendrum Studio",
      "name": "ZDS Shifter",
      "type": "input",
      "version": "",
      "state": "connected",
      "connection": "open"
    }
  ],
  "listeningDevices": ["-312145282"]
}
