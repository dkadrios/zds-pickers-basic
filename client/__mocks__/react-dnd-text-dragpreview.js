/* eslint-disable import/prefer-default-export, jsx-a11y/alt-text */
import React from 'react'

export const createDragPreview = () => <img src={null} />
