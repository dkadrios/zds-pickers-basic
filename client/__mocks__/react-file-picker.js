/* eslint-disable import/prefer-default-export */
export const FilePicker = () => <div>content</div>
