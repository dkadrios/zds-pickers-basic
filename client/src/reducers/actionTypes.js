export default {
  ADD_ENTRY: 'ADD_ENTRY',
  CANCEL_ENTRY_EDIT: 'CANCEL_ENTRY_EDIT',
  CHANGE_CONTROL_TYPE: 'CHANGE_CONTROL_TYPE',
  CHANGE_ENTRY_CHANNEL: 'CHANGE_ENTRY_CHANNEL',
  CHANGE_ENTRY_STATUS: 'CHANGE_ENTRY_STATUS',
  CHANGE_ENTRY_VALUE: 'CHANGE_ENTRY_VALUE',
  CHANGE_GROUP_CHANNEL: 'CHANGE_GROUP_CHANNEL',
  CHANGE_GROUP_VALUE: 'CHANGE_GROUP_VALUE',
  CHANGE_INPUT_CONTROL_CHANNEL: 'CHANGE_INPUT_CONTROL_CHANNEL',
  CHANGE_INPUT_CONTROL_LATCHING: 'CHANGE_INPUT_CONTROL_LATCHING',
  CHANGE_INPUT_CONTROL_POLARITY: 'CHANGE_INPUT_CONTROL_POLARITY',
  CHANGE_INPUT_CONTROL_VALUE: 'CHANGE_INPUT_CONTROL_VALUE',
  CHANGE_SELECTED_GROUP: 'CHANGE_SELECTED_GROUP',
  CHECKED_REGISTRATION: 'CHECKED_REGISTRATION',
  CLEAR_ALL_SELECTIONS: 'CLEAR_ALL_SELECTIONS',
  CONFIRM_FACTORY_RESET: 'CONFIRM_FACTORY_RESET',
  DELETE_MAPPING: 'DELETE_MAPPING',
  DEVICE_REGISTERED: 'DEVICE_REGISTERED',
  DISMISS_ERROR: 'DISMISS_ERROR',
  EDIT_ENTRY: 'EDIT_ENTRY',
  ENTRY_META_CLICKED: 'ENTRY_META_CLICKED',
  EXPORT_SETTINGS_PACKET: 'EXPORT_SETTINGS_PACKET',
  EXPORT_SETTINGS: 'EXPORT_SETTINGS',
  FACTORY_RESET: 'FACTORY_RESET',
  GET_SYSEX_CONTROLS: 'GET_SYSEX_CONTROLS',
  GET_SYSEX_GROUPS: 'GET_SYSEX_GROUPS',
  GET_SYSEX_MODEL: 'GET_SYSEX_MODEL',
  GET_SYSEX_VERSION: 'GET_SYSEX_VERSION',
  HARDWARE_TEST_FACTORY_RESET: 'HARDWARE_TEST_FACTORY_RESET',
  HARDWARE_TEST_NEXT: 'HARDWARE_TEST_NEXT',
  HARDWARE_TEST_SEND_ENTRIES: 'HARDWARE_TEST_SEND_ENTRIES',
  HARDWARE_TEST_SEND_MESSAGE: 'HARDWARE_TEST_SEND_MESSAGE',
  HARDWARE_TEST_SEND_MESSAGES: 'HARDWARE_TEST_SEND_MESSAGES',
  HIDE_HARDWARE_TEST_DIALOG: 'HIDE_HARDWARE_TEST_DIALOG',
  HIDE_REGISTRATION_DLG: 'HIDE_REGISTRATION_DLG',
  HIDE_REGISTRATION_NAG: 'HIDE_REGISTRATION_NAG',
  IMPORT_MAPPING: 'IMPORT_MAPPING',
  IMPORT_SETTINGS: 'IMPORT_SETTINGS',
  INVALID_SETTINGS_FILE_ACK: 'INVALID_SETTINGS_FILE_ACK',
  INVALID_SETTINGS_FILE: 'INVALID_SETTINGS_FILE',
  MAPPING_CHANGED: 'MAPPING_CHANGED',
  MIDI_IDLE: 'MIDI_IDLE',
  MIDI_IN_ACTIVITY: 'MIDI_IN_ACTIVITY',
  MIDI_OUT_ACTIVITY: 'MIDI_OUT_ACTIVITY',
  NOT_RESPONDING: 'NOT_RESPONDING',
  RECEIVED_AVAILABILITY: 'RECEIVED_AVAILABILITY',
  RECEIVED_CONTROLS: 'RECEIVED_CONTROLS',
  RECEIVED_ENTRY: 'RECEIVED_ENTRY',
  RECEIVED_GROUP: 'RECEIVED_GROUP',
  RECEIVED_GROUPS: 'RECEIVED_GROUPS',
  RECEIVED_MODEL: 'RECEIVED_MODEL',
  RECEIVED_STATE: 'RECEIVED_STATE',
  RECEIVED_VERSION: 'RECEIVED_VERSION',
  REMOVE_ENTRY: 'REMOVE_ENTRY',
  REPORT_ERROR: 'REPORT_ERROR',
  RESTART: 'RESTART',
  SAVE_ENTRY_EDIT: 'SAVE_ENTRY_EDIT',
  SEARCHED_FOR_SHIFTER: 'SEARCHED_FOR_SHIFTER',
  SELECT_ALL_ENTRIES: 'SELECT_ALL_ENTRIES',
  SELECT_SHIFT_ENTRY: 'SELECT_SHIFT_ENTRY',
  SET_FLAGS: 'SET_FLAGS',
  SHIFTER_FOUND: 'SHIFTER_FOUND',
  SHIFTER_MISSING: 'SHIFTER_MISSING',
  SHOW_EXPORT_DIALOG: 'SHOW_EXPORT_DIALOG',
  SHOW_HARDWARE_TEST_DIALOG: 'SHOW_HARDWARE_TEST_DIALOG',
  SHOW_IMPORT_DIALOG: 'SHOW_IMPORT_DIALOG',
  SHOW_MAPPINGS_DLG: 'SHOW_MAPPINGS_DLG',
  SHOW_REGISTRATION_DLG: 'SHOW_REGISTRATION_DLG',
  SHOW_USER_MAPPINGS_DIALOG: 'SHOW_USER_MAPPINGS_DIALOG',
  SORT_GROUP: 'SORT_GROUP',
  TEST_INTERFACE_FOUND: 'TEST_INTERFACE_FOUND',
  TEST_INTERFACE_MISSING: 'TEST_INTERFACE_MISSING',
}
