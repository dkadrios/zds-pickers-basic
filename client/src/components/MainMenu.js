import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import Divider from '@material-ui/core/Divider'
import MenuIcon from '@material-ui/icons/Menu'
import PermIdentity from '@material-ui/icons/PermIdentity'
import FileDownload from '@material-ui/icons/GetApp'
import FileUpload from '@material-ui/icons/CloudUpload'
import Warning from '@material-ui/icons/Warning'
import { actions as userActions } from '../reducers/user'
import { actions as shifterActions } from '../reducers/shifter'
import { shifterShape } from '../shapes'
import ExportSettings from './file/ExportSettings'
import ImportSettings from './file/ImportSettings'
import FactoryReset from './FactoryReset'

export class MainMenu extends Component {
  state = {
    anchorEl: null,
  }

  handleClick = (event) => {
    this.setState({ anchorEl: event.currentTarget })
  }

  handleClose = () => {
    this.setState({ anchorEl: null })
  }

  render() {
    const {
      confirmFactoryReset,
      performFactoryReset,
      shifter,
      shifter: { ready, found, responding },
      showDialog,
      showExportDialog,
      showImportDialog,
    } = this.props
    const { anchorEl } = this.state

    const open = Boolean(anchorEl)

    return ready ? (
      <>
        <IconButton
          aria-haspopup="true"
          aria-label="More"
          aria-owns={open ? 'long-menu' : null}
          onClick={this.handleClick}
        >
          <Tooltip
            title="Settings"
          >
            <MenuIcon />
          </Tooltip>
        </IconButton>
        <Menu
          anchorEl={anchorEl}
          id="long-menu"
          onClick={this.handleClose}
          onClose={this.handleClose}
          open={open}
          PaperProps={{
            style: {
              width: 200,
            },
          }}
        >
          <MenuItem
            icon="perm_identity"
            onClick={showDialog}
            tag="miRegistration"
            value="user"
          >
            <ListItemIcon>
              <PermIdentity />
            </ListItemIcon>
            <ListItemText
              inset
              primary="Registration"
            />
          </MenuItem>

          <Divider />

          <MenuItem
            disabled={!(found && responding)}
            onClick={() => showExportDialog(true)}
            tag="miExport"
          >
            <ListItemIcon>
              <FileDownload />
            </ListItemIcon>
            <ListItemText
              inset
              primary="Backup Settings"
            />
          </MenuItem>

          <MenuItem
            disabled={!(found && responding)}
            onClick={() => showImportDialog(true)}
            tag="miImport"
          >
            <ListItemIcon>
              <FileUpload />
            </ListItemIcon>
            <ListItemText
              inset
              primary="Restore Settings"
            />
          </MenuItem>

          <Divider />

          <MenuItem
            onClick={() => confirmFactoryReset(true)}
            tag="miReset"
            value="reset"
          >
            <ListItemIcon>
              <Warning />
            </ListItemIcon>
            <ListItemText
              inset
              primary="Factory Reset"
            />
          </MenuItem>
        </Menu>

        <ExportSettings />
        <ImportSettings />

        <FactoryReset
          confirmFactoryReset={confirmFactoryReset}
          performFactoryReset={performFactoryReset}
          shifter={shifter}
        />
      </>
    ) : null
  }
}

MainMenu.propTypes = {
  shifter: shifterShape.isRequired,
  showDialog: PropTypes.func.isRequired,
  confirmFactoryReset: PropTypes.func.isRequired,
  performFactoryReset: PropTypes.func.isRequired,
  showImportDialog: PropTypes.func.isRequired,
  showExportDialog: PropTypes.func.isRequired,
}

export const mapStateToProps = ({ shifter }) => ({ shifter })
export const mapDispatchToProps = dispatch => bindActionCreators({ ...userActions, ...shifterActions }, dispatch)

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(MainMenu)
