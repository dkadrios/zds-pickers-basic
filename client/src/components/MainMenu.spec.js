import { shifter } from '../../__mocks__'
import actionTypes from '../reducers/actionTypes'
import { MainMenu, mapDispatchToProps, mapStateToProps } from './MainMenu'

describe('MainMenu tests', () => {
  const showDialog = jest.fn()
  const confirmFactoryReset = jest.fn()
  const performFactoryReset = jest.fn()
  const showImportDialog = jest.fn()
  const showExportDialog = jest.fn()

  const props = {
    shifter,
    showDialog,
    confirmFactoryReset,
    performFactoryReset,
    showImportDialog,
    showExportDialog,
  }

  beforeEach(jest.resetAllMocks)

  it('renders correctly', () => {
    shallowExpect(MainMenu)(props).toMatchSnapshot()
  })

  it('clicking should open the menu', () => {
    const wrapper = shallow(<MainMenu {...props} />)
    wrapper.instance().handleClick({ currentTarget: 'target' })
    expect(wrapper.state('anchorEl')).toEqual('target')
  })

  it('clicking should hide the menu when closing', () => {
    const wrapper = shallow(<MainMenu {...props} />)
    wrapper.setState({ anchorEl: 'target' })
    wrapper.instance().handleClose()
    expect(wrapper.state('anchorEl')).toBeNull()
  })

  it('should show registration dialog', () => {
    const wrapper = shallow(<MainMenu {...props} />)
    wrapper
      .find('WithStyles(MenuItem)')
      .filter('[tag="miRegistration"]')
      .simulate('click')
    expect(showDialog).toHaveBeenCalled()
  })

  it('should show export dialog', () => {
    const wrapper = shallow(<MainMenu {...props} />)
    wrapper
      .find('WithStyles(MenuItem)')
      .filter('[tag="miExport"]')
      .simulate('click')
    expect(showExportDialog).toHaveBeenCalledWith(true)
  })

  it('should show import dialog', () => {
    const wrapper = shallow(<MainMenu {...props} />)
    wrapper
      .find('WithStyles(MenuItem)')
      .filter('[tag="miImport"]')
      .simulate('click')
    expect(showImportDialog).toHaveBeenCalledWith(true)
  })

  it('should show reset dialog', () => {
    const wrapper = shallow(<MainMenu {...props} />)
    wrapper
      .find('WithStyles(MenuItem)')
      .filter('[tag="miReset"]')
      .simulate('click')
    expect(confirmFactoryReset).toHaveBeenCalledWith(true)
  })

  it('should utilize mapStateToProps', () => {
    const state = { shifter }
    expect(mapStateToProps(state)).toEqual(state)
  })

  it('should utilize mapDispatchToProps', () => {
    const dispatch = jest.fn()
    mapDispatchToProps(dispatch).showDialog()
    expect(dispatch.mock.calls[0][0]).toEqual({ type: actionTypes.SHOW_REGISTRATION_DLG })
  })
})
