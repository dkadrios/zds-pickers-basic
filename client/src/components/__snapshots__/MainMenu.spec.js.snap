// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainMenu tests renders correctly 1`] = `
<Fragment>
  <WithStyles(IconButton)
    aria-haspopup="true"
    aria-label="More"
    aria-owns={null}
    onClick={[Function]}
  >
    <WithStyles(Tooltip)
      title="Settings"
    >
      <pure(MenuIcon) />
    </WithStyles(Tooltip)>
  </WithStyles(IconButton)>
  <WithStyles(Menu)
    PaperProps={
      Object {
        "style": Object {
          "width": 200,
        },
      }
    }
    anchorEl={null}
    id="long-menu"
    onClick={[Function]}
    onClose={[Function]}
    open={false}
  >
    <WithStyles(MenuItem)
      icon="perm_identity"
      onClick={[MockFunction]}
      tag="miRegistration"
      value="user"
    >
      <WithStyles(ListItemIcon)>
        <pure(PermIdentityIcon) />
      </WithStyles(ListItemIcon)>
      <WithStyles(ListItemText)
        inset={true}
        primary="Registration"
      />
    </WithStyles(MenuItem)>
    <WithStyles(Divider) />
    <WithStyles(MenuItem)
      disabled={false}
      onClick={[Function]}
      tag="miExport"
    >
      <WithStyles(ListItemIcon)>
        <pure(GetAppIcon) />
      </WithStyles(ListItemIcon)>
      <WithStyles(ListItemText)
        inset={true}
        primary="Backup Settings"
      />
    </WithStyles(MenuItem)>
    <WithStyles(MenuItem)
      disabled={false}
      onClick={[Function]}
      tag="miImport"
    >
      <WithStyles(ListItemIcon)>
        <pure(CloudUploadIcon) />
      </WithStyles(ListItemIcon)>
      <WithStyles(ListItemText)
        inset={true}
        primary="Restore Settings"
      />
    </WithStyles(MenuItem)>
    <WithStyles(Divider) />
    <WithStyles(MenuItem)
      onClick={[Function]}
      tag="miReset"
      value="reset"
    >
      <WithStyles(ListItemIcon)>
        <pure(WarningIcon) />
      </WithStyles(ListItemIcon)>
      <WithStyles(ListItemText)
        inset={true}
        primary="Factory Reset"
      />
    </WithStyles(MenuItem)>
  </WithStyles(Menu)>
  <Connect(ExportSettings) />
  <Connect(ReduxForm) />
  <FactoryReset
    confirmFactoryReset={[MockFunction]}
    performFactoryReset={[MockFunction]}
    shifter={
      Object {
        "accessGranted": true,
        "errorMessage": "",
        "errorVisible": false,
        "exportBuffer": Array [],
        "exportDialogVisible": false,
        "exportFilename": "zds-shifter-backup.txt",
        "found": true,
        "hardwareTestVisible": true,
        "importDialogVisible": false,
        "importInProcess": false,
        "invalidSettingsFile": null,
        "midiActivityLEDMode": 0,
        "midiInActivity": false,
        "midiOutActivity": false,
        "ready": true,
        "resetInProcess": false,
        "responding": true,
        "searchedForShifter": true,
        "serialMidiOutEnabled": true,
        "showResetDialog": false,
        "testInterfaceFound": false,
        "usbMidiOutEnabled": true,
      }
    }
  />
</Fragment>
`;
